src/shell/shell_cmds.cpp

```diff
@@ -1906,109 +1907,92 @@ void DOS_Shell::CMD_LOADHIGH(char *args){
-{
-	const char *last = strchr(args, 0);
-	StripSpaces(args);
-	temp = ScanCMDRemain(args);
-	const bool optC = temp && tolower(temp[1]) == 'c';
-	const bool optT = temp && tolower(temp[1]) == 't';
-	if (temp && *temp && !optC && !optT)
-		return false;
-	if (temp) {
-		if (args == temp)
-			args = strchr(temp, 0) + 1;
-		temp += 2;
-		if (temp[0] == ':')
-			temp++;
-	}
-	if (optC) {
-		rem = temp;
-	} else if (optT) {
-		if (temp && *temp && *(temp + 1) == ',') {
-			wait_char = *temp;
-			wait_sec = atoi(temp + 2);
-		} else
-			wait_sec = 0;
-	}
-	if (args > last)
-		args = NULL;
-	return true;
-}
-
 void MAPPER_AutoType(std::vector<std::string> &sequence,
                      const uint32_t wait_ms,
                      const uint32_t pacing_ms);
 void MAPPER_AutoType_End();
 
 void DOS_Shell::CMD_CHOICE(char * args){
 	HELP("CHOICE");
-	static char defchoice[3] = {'y','n',0};
-	char *rem = nullptr;
-	char *ptr = nullptr;
-	char *temp = nullptr;
-	char wait_char = '\0';
-	int wait_sec = 0;
-	bool optN = false;
-	bool optS = false;
-	if (args) {
-		optN = ScanCMDBool(args, "N");
-		optS = ScanCMDBool(args, "S"); // Case-sensitive matching
-		if (!get_param(args, rem, temp, wait_char, wait_sec)) {
-			WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"), temp);
-			return;
-		}
-		if (args && !get_param(args, rem, temp, wait_char, wait_sec)) {
-			WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"), temp);
-			return;
-		}
-	}
-	if (!rem || !*rem)
-		rem = defchoice; /* No choices specified use YN */
-	ptr = rem;
-	uint8_t c;
-	if (!optS) while ((c = *ptr)) *ptr++ = (char)toupper(c); /* When in no case-sensitive mode. make everything upcase */
-	if (args && *args ) {
-		StripSpaces(args);
-		size_t argslen = strlen(args);
-		if (argslen > 1 && args[0] == '"' && args[argslen-1] == '"') {
-			args[argslen-1] = 0; //Remove quotes
-			args++;
-		}
-		WriteOut(args);
-	}
-	/* Show question prompt of the form [a,b]? where a b are the choice values */
-	if (!optN) {
-		if (args && *args) WriteOut(" ");
-		WriteOut("[");
-		size_t len = strlen(rem);
-		for (size_t t = 1; t < len; t++) {
-			WriteOut("%c,",rem[t-1]);
-		}
-		WriteOut("%c]?",rem[len-1]);
-	}
 
-	std::vector<std::string> sequence;
-	bool in_char = optS ? (strchr(rem, wait_char) != nullptr)
-	                    : (strchr(rem, toupper(wait_char)) ||
-	                       strchr(rem, tolower(wait_char)));
-	bool auto_type = wait_char && *rem && in_char && wait_sec > 0;
-	if (auto_type) {
-		sequence.emplace_back(std::string(1, tolower(wait_char)));
-		MAPPER_AutoType(sequence, wait_sec * 1000, 500);
-	}
-	uint16_t n = 1;
-	do {
+	// Parse "/n"; does the user want to see the choices or not?
+	const bool wants_quiet = ScanCMDBool(args, "N");
+
+	// Parse "/s"; does the user want case-sensitive choices?
+	const bool wants_case_sensitive = ScanCMDBool(args, "S");
+
+	// Prepare the command line for use with regular expressions
+	assert(args);
```

Comment:
We're dealing with user supplied data, assert probably isn't the correct thing to do here. We should probably write a nice error message to the user.

Reply:
The command-line is handled upstream and regardless of user input, should always be impossible to get a null pointer in here (at a minimum, might be an empty string).